src/dbus/server/dbus_thread_object.cpp

```diff
@@ -556,5 +589,248 @@ otError DBusThreadObject::GetSupportedChannelMaskHandler(DBusMessageIter &aIter)
     return err;
 }
 
+otError DBusThreadObject::GetRloc16Handler(DBusMessageIter &aIter)
+{
+    auto     threadHelper = mNcp->GetThreadHelper();
+    otError  err          = OT_ERROR_NONE;
+    uint16_t rloc16       = otThreadGetRloc16(threadHelper->GetInstance());
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, rloc16) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+}
+
+otError DBusThreadObject::GetExtendedAddressHandler(DBusMessageIter &aIter)
+{
+    auto                threadHelper    = mNcp->GetThreadHelper();
+    otError             err             = OT_ERROR_NONE;
+    const otExtAddress *addr            = otLinkGetExtendedAddress(threadHelper->GetInstance());
+    uint64_t            extendedAddress = ConvertOpenThreadUint64(addr->m8, sizeof(addr->m8));
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, extendedAddress) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+}
+
+otError DBusThreadObject::GetRouterIdHandler(DBusMessageIter &aIter)
+{
+    auto         threadHelper = mNcp->GetThreadHelper();
+    otError      err          = OT_ERROR_NONE;
+    uint16_t     rloc16       = otThreadGetRloc16(threadHelper->GetInstance());
+    otRouterInfo info;
+
+    SuccessOrExit(err = otThreadGetRouterInfo(threadHelper->GetInstance(), rloc16, &info));
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, info.mRouterId) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+}
+
+otError DBusThreadObject::GetLeaderDataHandler(DBusMessageIter &aIter)
+{
+    auto                threadHelper = mNcp->GetThreadHelper();
+    otError             err          = OT_ERROR_NONE;
+    struct otLeaderData data;
+    otbrLeaderData      leaderData;
+
+    SuccessOrExit(err = otThreadGetLeaderData(threadHelper->GetInstance(), &data));
+    leaderData.mPartitionId       = data.mPartitionId;
+    leaderData.mWeighting         = data.mWeighting;
+    leaderData.mDataVersion       = data.mDataVersion;
+    leaderData.mStableDataVersion = data.mStableDataVersion;
+    leaderData.mLeaderRouterId    = data.mLeaderRouterId;
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, leaderData) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+}
+
+otError DBusThreadObject::GetNetworkDataHandler(DBusMessageIter &aIter)
+{
+    auto                 threadHelper = mNcp->GetThreadHelper();
+    otError              err          = OT_ERROR_NONE;
+    uint8_t              data[255]; // magic number from openthread/src/cli/cli.cpp:ProcessNetworkDataShow
+    uint8_t              len = sizeof(data);
+    std::vector<uint8_t> networkData;
+
+    otNetDataGet(threadHelper->GetInstance(), /*stable=*/false, data, &len);
```

Comment:
1. We can write to `networkData` directly.
2. Catch the return error.

Reply:
We cannot since we cannot really decide the length of the network data.